### PR TITLE
[viu] Fixed bypass method for preview/VIP content

### PR DESCRIPTION
**Pull request by zackmark29**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fix of bypassing/downloading of Preview (3mins) or VIP content with non-member.
https://github.com/yt-dlp/yt-dlp/issues/3246

### CODE REVIEW

File: yt_dlp/extractor/viu.py [hunk 1]
```diff
@@ -17,6 +16,7 @@
     smuggle_url,
     unsmuggle_url,
     url_or_none,
+    compat_urlparse,
```
Comment (pukkandan) [inline yt_dlp/extractor/viu.py:19]: use urllib.parse

Comment (zackmark29) [reply yt_dlp/extractor/viu.py:19]: Yeah that was I used first and thinking I'm doing wrong and not following the guide =D

File: yt_dlp/extractor/viu.py [hunk 2]
```diff
@@ -5,7 +5,11 @@
 import json
 import uuid
 import random
-import urllib.parse
+from urllib.parse import (
+    urlparse,
+    urlencode,
+    parse_qsl
+)
```
Comment (pukkandan) [inline yt_dlp/extractor/viu.py:12]:
```suggestion
import urllib.parse
```

File: yt_dlp/extractor/viu.py [hunk 3]
```diff
@@ -346,13 +351,13 @@ def download_playback():
 
             # bypass preview duration limit
             if duration_limit:
-                stream_url = urllib.parse.urlparse(stream_url)
+                old_stream_url = urlparse(stream_url)
+                query = dict(parse_qsl(old_stream_url.query, keep_blank_values=True))
```
Comment (pukkandan) [inline yt_dlp/extractor/viu.py:355]:
```suggestion
                old_stream_url = urllib.parse.urlparse(stream_url)
                query = dict(urllib.parse.parse_qsl(old_stream_url.query, keep_blank_values=True))
```

File: yt_dlp/extractor/viu.py [hunk 4]
```diff
@@ -346,13 +351,13 @@ def download_playback():
...
                 query.update({
                     'duration': video_data.get('time_duration') or '9999999',
                     'duration_start': '0',
                 })
-                stream_url = stream_url._replace(query=urllib.parse.urlencode(dict(
-                    urllib.parse.parse_qsl(stream_url.query, keep_blank_values=True)))).geturl()
+                stream_url = old_stream_url._replace(query=urlencode(query)).geturl()
```
Comment (pukkandan) [inline yt_dlp/extractor/viu.py:360]:
```suggestion
                stream_url = old_stream_url._replace(query=urllib.parse.urlencode(query)).geturl()
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Pull request merged**